src/main/ApplicationUtils.cpp

```diff
@@ -48,7 +48,18 @@ runWithConfig(Config cfg)
     Application::pointer app;
     try
     {
-        app = Application::create(clock, cfg, false);
+        if (Application::modeUsesInMemoryDB(mode))
```

Comment:
see my previous comment: those adjustments probably belong into the `Config` class for the most part where we'd have "settable flags" (modes) and "derived flags" (from the modes)